### PR TITLE
Updating docs and examples to match new structure of library

### PR DESCRIPTION
**Pull request by co9olguy**
**Description of the Change:** Updating docs and examples to be compliant with all new changes coming in version 0.11. 

This is also the final chance to consolidate any necessary name changes to new objects before release

**Benefits:** Consistency and clarity :)

**Possible Drawbacks:** None

**Related GitHub Issues:** #89, #108

### CODE REVIEW

**Comment by co9olguy** [comment]
@smite @josh146 Some other points that came up when revising docs, to make structure of library more consistent:

- [ ] put the `modes` argument from `eng.run` into `state_options` dictionary
- [ ] rename `state_options` dictionary to `result_options` or `return_options`
- [ ] rename `DeviceSpecs` to `CircuitSpecs`

File: strawberryfields/engine.py [hunk 1]
```diff
@@ -356,6 +356,8 @@ def run(self, program, *, shots=1, compile_options={}, modes=None, state_options
             Result: results of the computation
         """
 
+        # session or feed_dict are needed by TF backend during simulation if program contains measurements
+        kwargs.update(state_options)
```
Comment (co9olguy) [inline strawberryfields/engine.py]: Note: I needed to add this to ensure correct functionality of TF backend. It's not enough to pass these during the call to `state`

**Comment by co9olguy** [comment]
Outside of the checklist above (whose items are more of a refactor than a documentation fix), I think the documentation is now all up-to-date. 

Ready for review :+1:

File: doc/tutorials/tutorial_post_selection.rst [hunk 2]
```diff
@@ -291,20 +301,22 @@ For example, the above user defined ``log`` function can be rewritten using an e
...
+    prog = sf.Program(3)
+
+    with prog.context as q:
         MeasureX      | q[0]
         MeasureP      | q[1]
         Dgate(RR([q[0],q[1]], lambda q0,q1: q0+1j*q1)) | q[2]
 
-:html:`</div></div>`
\ No newline at end of file
+:html:`</div></div>`
```
Comment (co9olguy) [inline doc/tutorials/tutorial_post_selection.rst]: ??

Comment (co9olguy) [inline doc/tutorials/tutorial_post_selection.rst]:
```suggestion
```

File: examples/gaussian_cloning.py [hunk 3]
```diff
@@ -30,23 +31,24 @@
...
+
+for i in range(reps):
+    eng.reset()
+    results = eng.run(gaussian_cloning, modes=[0])
+    f[i] = results.state.fidelity_coherent([0.7+1.2j])
+    a[i] = results.state.displacement()
+
+print("Fidelity of cloned state:", np.mean(f))
+print("Mean displacement of cloned state:", np.mean(a))
+print("Mean covariance matrix of cloned state:", np.cov([a.real, a.imag]))
```
Comment (co9olguy) [inline examples/gaussian_cloning.py]:
```suggestion
print("Mean covariance matrix of cloned state:", np.cov([a.real, a.imag]))

```

**Comment by co9olguy** [comment]
> @smite @josh146 Some other points that came up when revising docs, to make structure of library more consistent:
> 
>     * [ ]  put the `modes` argument from `eng.run` into `state_options` dictionary
> 
>     * [ ]  rename `state_options` dictionary to `result_options`, `return_options`, or `run_options` (last one is my favourite, since TFBackend actually needs these options during execution)
> 
>     * [ ]  rename `DeviceSpecs` to `CircuitSpecs`

I'm thinking these naming refactors should be spun out as a separate PR to the docs one (should be final task before release)

File: doc/gallery/gallery.rst [hunk 4]
```diff
@@ -124,12 +129,30 @@ Research papers
 
 Finally, some links to studies and research papers that utilize Strawberry Fields.
 
-#. D\. Su, K. K. Sabapathy, C. R. Myers, H. Qi, C. Weedbrook, and K. Brádler. *Implementing quantum algorithms on temporal photonic cluster states.* arXiv, 2018. `arXiv:1805.02645 <https://arxiv.org/abs/1805.02645>`_.
+#. D\. Su, K. K. Sabapathy, C. R. Myers, H. Qi, C. Weedbrook, and K. Brádler. *Implementing quantum algorithms on temporal photonic cluster states.* `Physical Review A 98, 032316 <https://journals.aps.org/pra/abstract/10.1103/PhysRevA.98.032316>`_, 2018.
```
Comment (smite) [inline doc/gallery/gallery.rst]: These could be included in `references.bib` and listed automatically using sphinxcontrib-bibtex.

File: doc/tutorials/tutorial_post_selection.rst (hunk 2 above)
Comment (co9olguy) [reply doc/tutorials/tutorial_post_selection.rst]: seems like this was put in to make spacing consistent between different pages. I'm going to leave it in

**Pull request merged**